pandas/plotting/_matplotlib/core.py

```diff
@@ -423,13 +425,19 @@ def _compute_plot_data(self):
         if is_empty:
             raise TypeError("no numeric data to plot")
 
-        # GH25587: cast ExtensionArray of pandas (IntegerArray, etc.) to
-        # np.ndarray before plot.
-        numeric_data = numeric_data.copy()
-        for col in numeric_data:
-            numeric_data[col] = np.asarray(numeric_data[col])
+        def convert_to_ndarray(data):
```

Comment:
you could use `pd.to_numeric` here I think. Illustrative example:
``` python
In [8]: s = pd.Series([1, 2, 3, pd.NA], dtype="Int64")

In [9]: pd.to_numeric(s.to_numpy(), downcast="integer")
Out[9]: array([ 1.,  2.,  3., nan])
```

Reply:
Thank you @arw2019 for the example. I think I didn't get your point here. Can you please explain it a bit more?

Reply:
no don't use `pd.to_numeric` we already know the type and its an internal function